src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -84,8 +90,9 @@
         String[] fileDiffResultList = fullDiffResult.split(DIFF_FILE_CHUNK_SEPARATOR);
 
         for (String fileDiffResult : fileDiffResultList) {
-            // file deleted or is a binary file, skip it
-            if (fileDiffResult.startsWith(FILE_DELETED_METADATA) || fileDiffResult.contains(BINARY_FILE_SYMBOL)) {
+            // file deleted, is binary file or is new and empty file, skip it
+            if (FILE_DELETED_PREDICATE.test(fileDiffResult) || fileDiffResult.contains(BINARY_FILE_SYMBOL)
```

Comment:
Wouldn't this fail if you replace `startsWith` with predicate test?

Reply:
>Wouldn't this fail if you replace startsWith with predicate test?

The predicate has `^` at the start to indicate that the `String` must start with this pattern, thus its the same as `startsWith`, just that `startWith` must be an exact `String`, while we need to check for  a pattern.